internal/test/helpers.go

```diff
@@ -249,7 +248,5 @@ func GenerateCIArtifactName(prefix string) (string, error) {
 		return "", fmt.Errorf("The environment variables must specify BRANCH_NAME, BUILD_ID, and DISTRO_CODE")
 	}
 
-	arch := common.CurrentArch()
-
-	return fmt.Sprintf("%s%s-%s-%s-%s", prefix, distroCode, arch, branchName, buildId), nil
+	return fmt.Sprintf("%s%s-%s-%s", prefix, distroCode, branchName, buildId), nil
```

Comment:
won't this lead to conflicts once we start testing on multiple architectures ?

Reply:
Why not prefix all test artifacts with "testing_" and query the cloud environments directly like we did in the lorax days ?

Reply:
What I did in this PR is that I use the Gitlab runner name as the `DISTRO_CODE`. That contains both the OS, version and architecture and it's easy to recreate for the cloud-cleaner to use. I pretty much merged the variables. It's probably not ideal but it worked nicely in this case. 

Prefixing all artifacts with "testing_" also sounds good, I'm just not sure how much of a change in cloud-cleaner code would that require...

Reply:
Prefixing all artifacts with `testing_` won't work at least for GCP, because resource names are pretty much at the limit of their length. For GCP, I didn't use the content of the environment variables, but their hash to prevent issues with invalid characters in the resource name (e.g. `_` is not allowed). See https://github.com/osbuild/osbuild-composer/blob/c09638efe51d40b5ae1744fcbf71e5e22ff37dc9/test/cases/api.sh#L448

Reply:
FTR I proposed using tags, I hope that all cloud envs do support them. AWS and Azure surely do, what about GCP @thozza?

Reply:
GCP definitely has some features similar to tags, which we could use. AFAIK for Cloud Storage objects, we can set only custom metadata (key-value). We actually already set it to the desired image name in composer when we upload the built image to Storage, otherwise we could not possibly determine the association between storage objects and image names in `cloud-cleaner` https://github.com/osbuild/osbuild-composer/blob/b7df6ac6275a317d9741e981ac84872d353989c4/cmd/cloud-cleaner/main.go#L98

For GCP Compute resources, there are "tags", but these have a bit different use case and seem to be used only for networking. We can also set custom metadata, but probably the best approach would be to set ["Labels"](https://cloud.google.com/compute/docs/labeling-resources), which can be then used directly with `gcloud` or API calls to list only specific resources.

I must say that GCP is a bit confusing in the sense that each "product" (e.g. Cloud Storage vs. Compute Engine) have a different feature set WRT tags/labels/metadata.

Reply:
@ondrejbudai, @thozza can we mark this comment as resolved or do you have strong preference on using tags instead of naming conventions ?

Reply:
Hi @atodorov. I'm OK with using naming convention as long as it works with all cloud providers and existing test cases which are creating cloud resources.